### PR TITLE
Add a dev container to facilitate easy development environment

### PR DESCRIPTION
**Pull request by markspec**
Add a dev container to facilitate development and provide a conistent python/poetry/nox environment for developers.

### CODE REVIEW

**Review by tasansal — CHANGES_REQUESTED** [review]

File: notebooks/ingestion_and_export.ipynb
[the remark this answers is not in the document]
Comment (markspec) [reply notebooks/ingestion_and_export.ipynb:1]: I intentionally included the notebook.  As there is no constructor in mdio, debugging the export is a long process.  This notebook walks through the process required to test the export:

- Create segy
- Convert segy to mdio
- Check mdio file
- export mdio to segy 

It also demos the use of some debugging tools that may be beneficial to future developers.

File: docs/development_env.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Development Environment
+
+To facilitate development on different systems a [dev container](https://containers.dev/) has been added. This should seamlessly enable development for users of [VSCode](https://code.visualstudio.com/docs/devcontainers/containers) on systems with docker installed.
+
+For contributing guidelines please look here [link](../CONTRIBUTING.md)
+
+### known issues:
```
[the remark this answers is not in the document]
Comment (markspec) [reply docs/development_env.md]: Fixed in latest commit.

**Review by tasansal — CHANGES_REQUESTED** [review]

File: docs/development_env.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Development Environment
+
+To facilitate development on different systems a [dev container](https://containers.dev/) has been added. This should seamlessly enable development for users of [VSCode](https://code.visualstudio.com/docs/devcontainers/containers) on systems with docker installed.
+
+For contributing guidelines please look here [link](../CONTRIBUTING.md)
+
+### Known issues:
```
Comment (tasansal) [inline docs/development_env.md:7]:
```suggestion
### Known Issues:
```

File: docs/development_env.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Development Environment
+
+To facilitate development on different systems a [dev container](https://containers.dev/) has been added. This should seamlessly enable development for users of [VSCode](https://code.visualstudio.com/docs/devcontainers/containers) on systems with docker installed.
+
+For contributing guidelines please look here [link](../CONTRIBUTING.md)
+
+### Known Issues:
+
+- `git config --global --add safe.directory \`pwd\` ` might be needed inside the container.
```
Comment (tasansal) [inline docs/development_env.md:9]:
```suggestion
- `git config --global --add safe.directory $(pwd)` might be needed inside the container.
```
This fixes incorrect Markdown rendering

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -37,12 +37,16 @@ Request features on the [Issue Tracker].
...
-You need Python 3.7+ and the following tools:
+You need Python 3.9+ and the following tools:
 
 - [Poetry]
 - [Nox]
 - [nox-poetry]
 
+A [dev container](./docs/development_env.md) has been setup to provide a clean environment with the required dependencies.
+
+## How to install and run MDIO
```
Comment (tasansal) [inline CONTRIBUTING.md:48]:
```suggestion
## How to Install and Run MDIO
```

File: notebooks/ingestion_and_export.ipynb
[the remark this answers is not in the document]
Comment (markspec) [reply notebooks/ingestion_and_export.ipynb:1]: Removed

**Review by tasansal — APPROVED** [review]

**Pull request merged**